code/datums/extensions/health/_health.dm

```diff
@@ -0,0 +1,175 @@
+/*
+ * General standardized health processing for all atoms.
+ * Current health and max health are always rounded to the nearest whole number.
+ */
+/datum/extension/health
+	base_type = /datum/extension/health
+	expected_type = /atom
+
+	/// Current health, from `0` to `max_health`.
+	var/current_health
+
+	/// Maximum health. `0` or `NULL` means no health handling.
+	var/max_health
+
+	/// Whether or not the death state has been set.
+	var/death_state = FALSE
+
+	/// LAZY List of damage type resistance or weakness multipliers, decimal form. Only applied to health reduction.
+	var/list/damage_resistances
+
+	/// Minimum damage required to actually affect health in `can_damage_health()`
+	var/min_damage_threshhold = 0
+
+
+/**
+ * Retrieves the extension's `current_health`.
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/current_health()
+	return current_health
+
+
+/**
+ * Retrieves the extension's `max_health`.
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/max_health(real = FALSE)
+	var/atom/A = holder
+	return real ? max_health : A.get_max_health()
+
+
+/**
+ * Retrieves the extension's damage (Difference between `max_health` and `current_health`).
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/current_damage(damage_type = null)
+	return max_health - current_health
+
+
+/**
+ * Retrieves the extension's death state.
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/get_death_state()
+	return death_state
+
+
+/**
+ * Sets a new death_state.
+ * Returns `TRUE` of the new state is different from the old state, `FALSE` otherwise.
+ * Also calls `handle_death_change()` on the holder.
+ */
+/datum/extension/health/proc/set_death_state(new_death_state)
+	if (death_state == new_death_state)
+		return FALSE
+	death_state = new_death_state
+	var/atom/A = holder
+	A.handle_death_change(new_death_state)
+	return TRUE
+
+
+/**
+ * Checks the current condition of health and sets the death state accordingly.
+ * Returns `TRUE` if `death_state` was changed, `FALSE` otherwise.
+ */
+/datum/extension/health/proc/update_death_state()
+	if (current_health() == 0)
+		return set_death_state(TRUE)
+	else
+		return set_death_state(FALSE)
+
+
+/**
+ * Modifies `current_health` by the given modifier.
+ * Returns `TRUE` if `death_state` was changed, `FALSE` otherwise.
+ */
+/datum/extension/health/proc/mod_health(health_mod, damage_type = null)
```

Comment:
`mod` has a pretty specific meaning in mathematics (modulo) and another meaning in Neb's context (modpack) - `adjust_health` would line up with the other damage procs and avoid ambiguity.